lib/gpu/lib/src/render_pass.dart

```diff
@@ -150,6 +150,20 @@ base class SamplerOptions {
   SamplerAddressMode heightAddressMode;
 }
 
+base class Scissor {
+  Scissor({this.x = 0, this.y = 0, this.width = 0, this.height = 0}) {
+    _validate();
+  }
+
+  final int x, y, width, height;
```

Comment:
Renderers will often want to iterate on the scissor bounds over time between draws, so allowing all of these properties to be mutable would be more convenient. Otherwise the user will have to make a mirror data structure to do so.

Reply:
Done. I have one doubt here though, validate is a private function of Scissor, how is it accessible outside the class?